src/index.ts

```diff
@@ -113,19 +129,36 @@ export class Steveo implements ISteveo {
   }
 
   runner() {
-    return runner(
-      this.config.engine,
-      this.config,
-      this.registry,
-      this.pool,
-      this.logger,
-      this.hooks
-    );
+    if (!this._runner) {
+      this._runner = runner(
+        this.config.engine,
+        this.config,
+        this.registry,
+        this.pool,
+        this.logger,
+        this.hooks
+      );
+    }
+    return this._runner;
+  }
+
+  adminClient() {
+    if (this.config.engine !== 'kafka') {
+      throw new Error(
+        'Admin client is only meant to be used with kafka, please use "steveo.runner().createQueues()" for other engines'
```

Comment:
@arshu16 yeah as per my note about, not sure i'm a fan of this pattern and exposing different interface based on backend.